build_thirdparty.sh

```diff
@@ -39,3 +39,32 @@ set -x
 
 # shellcheck disable=SC2086
 python3 "$YB_THIRDPARTY_DIR/python/yugabyte_db_thirdparty/yb_build_thirdparty_main.py" "$@"
+
+rc=$?
+
+set +xue
+if [[ -n "$SNYK_TOKEN" ]]; then
+  echo "Running Snyk Vulnerability scan"
+  if [[ $OSTYPE == linux* ]]; then
+    curl https://static.snyk.io/cli/latest/snyk-linux -o snyk
+  else
+    curl https://static.snyk.io/cli/latest/snyk-macos -o snyk
+  fi
+  chmod +x ./snyk
+  homedir="${BASH_SOURCE[0]%/*}"
+
+  ./snyk auth "${SNYK_TOKEN}"
+  rc2=$?
+  
+  if [[ $rc2 -ne 0 ]]; then
+    exit $rc
+  fi
```

Comment:
This is exactly the kind of code that we don't need when running in the `set -e` mode. I think it is better to have exit-on-error by default.